java.md

```diff
@@ -26,11 +26,21 @@ performance on workloads with light to moderate lock-contention: improved spin-l
 enhanced implementation of `Thread.onSpinWait()` on Graviton2.
 
 ### Java JVM Options
-There are numerous options that control the JVM and may lead to better performance. Three that
-have shown large (1.5x) improvements in some Java workloads are eliminating tiered compilation
-and restricting the size of the code cache which allows the Graviton2 cores to better predict
-branches. These are helpful on some workloads but can hurt on others so testing with and without
-them is essential: `-XX:-TieredCompilation -XX:ReservedCodeCacheSize=64M -XX:InitialCodeCacheSize=64M`.
+There are numerous options that control the JVM and may lead to better performance.
+
+- Three flags `-XX:-TieredCompilation -XX:ReservedCodeCacheSize=64M -XX:InitialCodeCacheSize=64M`
+have shown large (1.5x) improvements in some Java workloads.
+The flags are eliminating tiered compilation and are restricting the size of
+the code cache which allows the Graviton2 cores to better predict branches.
+These are helpful on some workloads but can hurt on others so testing with and without
+them is essential.
+
+- Crypto algorithm AES/GCM used by TLS has been optimized for Graviton.
+On Graviton2 GCM encrypt/decrypt
+[performance improves by 3.5x to 5x](https://github.com/aws/aws-graviton-getting-started/issues/110#issuecomment-989442948).
+The optimization is enabled by default in Corretto and OpenJDK 18 and later.
+The optimization has been backported to Corretto and OpenJDK 11 and 17
+and can be enabled with the flags `-jvmArgs '-XX:+UnlockDiagnosticVMOptions -XX:+UseAESCTRIntrinsics'`.
```

Comment:
do you need -jvmArgs? also i'm in the end is it behind UnlockDiagnosticVMOptions and doesn't work without it?

Reply:
The -jvmArgs looks like a carry-over from being run via some wrapper. I think that part can be dropped.

Reply:
I got the -jvmArgs flag from the command line in Xin Liu's experiment: https://github.com/aws/aws-graviton-getting-started/issues/110#issuecomment-989442948
Also both Xin's experiment and Liz Fong-Jones' blog https://aws.amazon.com/blogs/developer/tuning-apache-kafka-and-confluent-platform-for-graviton2-using-amazon-corretto/ mention both flags together.

Reply:
Xin has extracted the benchmarks here: https://github.com/navyxliu/openjdk_AESBench

Without -jvmArgs it fails to recognize the flags:
```
$ java -jar ./target/benchmarks.jar -XX:+UnlockDiagnosticVMOptions -XX:+UseAESCTRIntrinsics org.openjdk.bench.javax.crypto.small.AESGCMBench.*
Error parsing command line:
 'X' is not a recognized option
```
without the other option I also get an error:
```
Error: VM option 'UseAESCTRIntrinsics' is diagnostic and must be enabled via -XX:+UnlockDiagnosticVMOptions.
```

Reply:
You need ```-XX:+UnlockDiagnosticVMOptions```, but the ```-jvmArgs``` part is only for JMH or another wrapper, that is not a JVM argument itself.

Reply:
Makes sense. Thanks William for the explanation.  I pushed another update.